### ISSUE TITLE
Support real syntax validation instead of proxy metrics

### ISSUE THREAD

**Issue by timlawrenz** [timlawrenz/ratiocinator#5]
## Problem

During the GNN study, the heuristic `syntactic_validity_pct` metric (parenthesis matching + type distribution check) reported 99.5% validity for a model that produced **0% actually parseable Ruby code**. The metric was misleading enough to initially appear as a research success before manual qualitative analysis revealed the truth.

The orchestrator has no concept of "post-hoc validation" — it trusts whatever metrics the training script outputs.

## Proposed Solution

Add a `validation` section to ExperimentSpec:

```yaml
validation:
  command: "python validate_output.py --output-dir /workspace/output"
  timeout_s: 120
  required_metrics:
    - real_validity_pct  # Must appear in METRICS output
```

### Behavior

1. After training completes, run the validation command as a separate step
2. Validation script outputs its own `METRICS:{...}` line
3. These metrics are merged into (and can override) the training metrics
4. Allows real parsers (e.g., `ruby -c`, AST comparison) to validate outputs
5. Training metrics and validation metrics are clearly labeled in ResultStore

### Why This Matters

- Training scripts optimize proxy metrics; validation scripts measure ground truth
- Separation of concerns: the training script author doesn't need to embed a full parser
- Makes it impossible for a model to "game" a heuristic metric without detection
- The ResearchCoordinator's analysis phase gets accurate data to decide whether to iterate

## Files

- `src/ratiocinator/fleet/spec.py` — add `ValidationSpec` to `ExperimentSpec`
- `src/ratiocinator/fleet/executor.py` — add validation step after training in `_run_arm()`
- `src/ratiocinator/fleet/results.py` — label training vs validation metrics
